lib/ngx/re.lua

```diff
@@ -0,0 +1,163 @@
+-- I hereby assign copyright in this code to the lua-resty-core project,
+-- to be licensed under the same terms as the rest of the code.
+
+local ffi = require 'ffi'
+local bit = require "bit"
+local base = require "resty.core.base"
+local core_regex = require "resty.core.regex"
```

Comment:
Sounds good. For now should I re-implement `re_match_compile` (it'll be quite the copy-pasta, that one is huge) and `destroy_compiled_regex` in `ngx.re` then?